pkg/upgrade/upgrademanager/upgrademanager_test: TestConcurrentMigrationAttempts failed

pkg/upgrade/upgrademanager/upgrademanager_test.TestConcurrentMigrationAttempts [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9513246?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9513246?buildTab=artifacts#/) on master @ [85e41ca8d3d9edacf5ee3061a2591b159b9b0502](https://github.com/cockroachdb/cockroach/commits/85e41ca8d3d9edacf5ee3061a2591b159b9b0502):


```
=== RUN   TestConcurrentMigrationAttempts
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/0a595abbc53f575a0f489dc4e6e5da0f/logTestConcurrentMigrationAttempts1578974772
    test_log_scope.go:79: use -show-logs to present logs inline
    manager_external_test.go:402: 
        	Error Trace:	pkg/upgrade/upgrademanager/upgrademanager_test_test/pkg/upgrade/upgrademanager/manager_external_test.go:402
        	Error:      	Expected nil, but got: &pq.Error{Severity:"ERROR", Code:"42P01", Message:"write-job-info-delete: relation \"system.job_info\" does not exist", Detail:"", Hint:"", Position:"", InternalPosition:"", InternalQuery:"", Where:"", Schema:"", Table:"", Column:"", DataTypeName:"", Constraint:"", File:"errors.go", Line:"161", Routine:"NewUndefinedRelationError"}
        	Test:       	TestConcurrentMigrationAttempts
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0a595abbc53f575a0f489dc4e6e5da0f/logTestConcurrentMigrationAttempts1578974772
--- FAIL: TestConcurrentMigrationAttempts (5.26s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-schema
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestConcurrentMigrationAttempts.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-26756

pkg/upgrade/upgrademanager/upgrademanager_test.TestConcurrentMigrationAttempts [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9524747?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9524747?buildTab=artifacts#/) on master @ [f59767cd3201882102f596ba55de67f4e045409e](https://github.com/cockroachdb/cockroach/commits/f59767cd3201882102f596ba55de67f4e045409e):

Fatal error:

```
panic: test timed out after 4m55s

```
Stack: 

```
goroutine 67732 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestConcurrentMigrationAttempts
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/be5347fadc06714556335e3ce75bfe99/logTestConcurrentMigrationAttempts1207221848
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestConcurrentMigrationAttempts.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/upgrade/upgrademanager/upgrademanager_test.TestConcurrentMigrationAttempts [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9525011?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9525011?buildTab=artifacts#/) on master @ [b36c119c20c310cffde9b3311173dc4b8bee6ff0](https://github.com/cockroachdb/cockroach/commits/b36c119c20c310cffde9b3311173dc4b8bee6ff0):

Fatal error:

```
panic: test timed out after 4m55s

```
Stack: 

```
goroutine 66969 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestConcurrentMigrationAttempts
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/be5347fadc06714556335e3ce75bfe99/logTestConcurrentMigrationAttempts3665295436
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestConcurrentMigrationAttempts.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/upgrade/upgrademanager/upgrademanager_test.TestConcurrentMigrationAttempts [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9526286?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9526286?buildTab=artifacts#/) on master @ [bd2eaebb7e40b2479caf2c6f38a153e8d7cbf764](https://github.com/cockroachdb/cockroach/commits/bd2eaebb7e40b2479caf2c6f38a153e8d7cbf764):

Fatal error:

```
panic: test timed out after 4m55s

```
Stack: 

```
goroutine 69038 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestConcurrentMigrationAttempts
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/be5347fadc06714556335e3ce75bfe99/logTestConcurrentMigrationAttempts1051899325
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestConcurrentMigrationAttempts.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

another one here  https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/9528455?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

```
Failed
=== RUN   TestConcurrentMigrationAttempts
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/0a595abbc53f575a0f489dc4e6e5da0f/logTestConcurrentMigrationAttempts1131059060
    test_log_scope.go:79: use -show-logs to present logs inline
    manager_external_test.go:402:
          Error Trace:  pkg/upgrade/upgrademanager/upgrademanager_test_test/pkg/upgrade/upgrademanager/manager_external_test.go:402
          Error:        Expected nil, but got: &pq.Error{Severity:"ERROR", Code:"42P01", Message:"write-job-info-delete: relation \"system.job_info\" does not exist", Detail:"", Hint:"", Position:"", InternalPosition:"", InternalQuery:"", Where:"", Schema:"", Table:"", Column:"", DataTypeName:"", Constraint:"", File:"errors.go", Line:"161", Routine:"NewUndefinedRelationError"}
          Test:         TestConcurrentMigrationAttempts
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0a595abbc53f575a0f489dc4e6e5da0f/logTestConcurrentMigrationAttempts1131059060
--- FAIL: TestConcurrentMigrationAttempts (6.31s)
```

--- a/pkg/upgrade/upgrademanager/BUILD.bazel
+++ b/pkg/upgrade/upgrademanager/BUILD.bazel
@@ -63,7 +63,6 @@ go_test(
         "//pkg/sql/isql",
         "//pkg/testutils",
         "//pkg/testutils/serverutils",
-        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
         "//pkg/upgrade",

--- a/pkg/upgrade/upgrademanager/manager_external_test.go
+++ b/pkg/upgrade/upgrademanager/manager_external_test.go
@@ -31,7 +31,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfra"
 	"github.com/cockroachdb/cockroach/pkg/sql/isql"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/upgrade"
@@ -323,11 +322,10 @@ func TestConcurrentMigrationAttempts(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 101021, "flaky test")
 	// We're going to be migrating from the BinaryMinSupportedVersion to imaginary future versions.
 	current := clusterversion.TestingBinaryMinSupportedVersion
 	versions := []roachpb.Version{current}
-	for i := int32(1); i <= 5; i++ {
+	for i := int32(1); i <= 4; i++ {
 		v := current
 		v.Internal += i * 2
 		versions = append(versions, v)

